# Préparation des données

## Importation des librairies "Data Science" pour Python

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

#Chargement des différentes librairies

## Importation des données

**Données sur la population mondiale**

In [3]:
InputPopulation = pd.read_csv('Data/population.csv')
InputPopulation.head()

#Importation du fichier csv "population" et création du data frame

Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

**Aide alimentaire reçue (en tonnes) par pays, année et type de produit**

In [4]:
InputAideAlimentaire = pd.read_csv('Data/aide_alimentaire.csv')
InputAideAlimentaire.head()

#Importation du fichier csv "aide_alimentaire" et création du data frame

Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504

**Nombre de personnes en sous-alimentation (en millions d'habitants) par pays et intervalle d'années**

In [5]:
InputSousNutrition = pd.read_csv('Data/sous_nutrition.csv')
InputSousNutrition.head()

#Importation du fichier csv "sous_nutrition" et création du data frame

Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5

**Disponibilité Alimentaire par pays et type de produit pour l'année 2013**

In [6]:
InputDispoAlimentaire = pd.read_csv('Data/dispo_alimentaire.csv')
InputDispoAlimentaire.head()

#Importation du fichier csv "dispo_alimentaire" et création du data frame

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

## Création des Output

In [7]:
population = InputPopulation.copy()
AideAlimentaire = InputAideAlimentaire.copy()
SousNutrition = InputSousNutrition.copy()
DispoAlimentaire = InputDispoAlimentaire.copy()

#Copie des fichiers d'origine avec de conserver les originaux

## Analyse des données

### Population

In [8]:
population.shape

#Visualisation des dimensions du fichier "population"

(1416, 3)

In [9]:
population.dtypes

#Visualisation des types de données du fichier "population"

Zone       object
Année       int64
Valeur    float64
dtype: object

In [10]:
population[population.duplicated()].head()

#Détection des doublons

Empty DataFrame
Columns: [Zone, Année, Valeur]
Index: []

In [11]:
population.describe()

#Détection des outliers

Année        Valeur
count  1416.000000  1.416000e+03
mean   2015.500000  3.144793e+04
std       1.708428  1.300812e+05
min    2013.000000  7.930000e-01
25%    2014.000000  3.783410e+02
50%    2015.500000  5.126480e+03
75%    2017.000000  1.930666e+04
max    2018.000000  1.427648e+06

In [12]:
population.isnull().sum()

#Vérification des valeurs manquantes

Zone      0
Année     0
Valeur    0
dtype: int64

**Conclusion:**

- 1416 lignes, 3 colonnes
- Les types de données sont corrects
- Pas de doublons
- Pas de valeurs manquantes

**A corriger:**

- Renommer les colonnes "Zone" et "Valeur"
- Convertir les unités de population en millions d'habitants ( x 1000)

### Aide Alimentaire

In [13]:
AideAlimentaire.shape

#Visualisation des dimensions du fichier "aide_alimentaire"

(1475, 4)

In [14]:
AideAlimentaire.dtypes

#Visualisation des types de données du fichier "aide_alimentaire"

Pays bénéficiaire    object
Année                 int64
Produit              object
Valeur                int64
dtype: object

In [15]:
AideAlimentaire[AideAlimentaire.duplicated()].head()

#Détection des doublons

Empty DataFrame
Columns: [Pays bénéficiaire, Année, Produit, Valeur]
Index: []

In [16]:
AideAlimentaire.describe()

#Détection des outliers

Année         Valeur
count  1475.000000    1475.000000
mean   2014.054237    7481.966780
std       0.946916   23339.531424
min    2013.000000       0.000000
25%    2013.000000     287.500000
50%    2014.000000    1178.000000
75%    2015.000000    4334.500000
max    2016.000000  265013.000000

In [17]:
AideAlimentaire.isnull().sum()

#Vérification des valeurs manquantes

Pays bénéficiaire    0
Année                0
Produit              0
Valeur               0
dtype: int64

**Conclusion:**

- 1475 lignes, 4 colonnes
- Les types de données sont corrects
- Pas de doublons
- Pas de valeurs manquantes

**A corriger:**

- Renommer les colonnes "Pays bénéficiaire" et "Valeur"

### Sous Nutrition

In [18]:
SousNutrition.shape

#Visualisation des dimensions du fichier "sous_nutrition"

(1218, 3)

In [19]:
SousNutrition.dtypes

#Visualisation des types de données du fichier "sous_nutrition"

Zone      object
Année     object
Valeur    object
dtype: object

In [20]:
SousNutrition[SousNutrition.duplicated()].head()

#Détection des doublons

Empty DataFrame
Columns: [Zone, Année, Valeur]
Index: []

In [21]:
SousNutrition.describe()

#Détection des outliers

Zone      Année Valeur
count          1218       1218    624
unique          203          6    139
top     Afghanistan  2012-2014   <0.1
freq              6        203    120

In [22]:
SousNutrition.loc[SousNutrition['Valeur'] == '<0.1']

# 120 lignes avec pour valeur '<0.1'

Zone      Année Valeur
60    Arménie  2012-2014   <0.1
61    Arménie  2013-2015   <0.1
62    Arménie  2014-2016   <0.1
63    Arménie  2015-2017   <0.1
64    Arménie  2016-2018   <0.1
...       ...        ...    ...
1183  Vanuatu  2013-2015   <0.1
1184  Vanuatu  2014-2016   <0.1
1185  Vanuatu  2015-2017   <0.1
1186  Vanuatu  2016-2018   <0.1
1187  Vanuatu  2017-2019   <0.1

[120 rows x 3 columns]

In [23]:
SousNutrition.isnull().sum()

#Vérification des valeurs manquantes

Zone        0
Année       0
Valeur    594
dtype: int64

**Conclusion:**

- 1218 lignes, 3 colonnes
- dtype de la colonne 'valeur' en object !
- Pas de doublons
- 594 valeurs manquantes dans la colonne 'Valeur'
- 120 lignes avec "<0.1" dans la colonne 'Valeur'


**A corriger:**

- Corriger le dtype de la colonne valeur en float
- Remplacer les valeurs manquantes par 0
- Remplacer les "<0.1" par 0,01
- Convertir les unités de population en millions d'habitants ( x 1 000 000)
- Renommer les colonnes "Zone" et "Valeur"

### Dispo Alimentaire

In [24]:
DispoAlimentaire.shape

#Visualisation des dimensions du fichier "dispo_alimentaire"

(15605, 18)

In [25]:
DispoAlimentaire.dtypes

#Visualisation des types de données du fichier "dispo_alimentaire"

Zone                                                              object
Produit                                                           object
Origine                                                           object
Aliments pour animaux                                            float64
Autres Utilisations                                              float64
Disponibilité alimentaire (Kcal/personne/jour)                   float64
Disponibilité alimentaire en quantité (kg/personne/an)           float64
Disponibilité de matière grasse en quantité (g/personne/jour)    float64
Disponibilité de protéines en quantité (g/personne/jour)         float64
Disponibilité intérieure                                         float64
Exportations - Quantité                                          float64
Importations - Quantité                                          float64
Nourriture                                                       float64
Pertes                                             

In [26]:
DispoAlimentaire[DispoAlimentaire.duplicated()].head()

#Détection des doublons

Empty DataFrame
Columns: [Zone, Produit, Origine, Aliments pour animaux, Autres Utilisations, Disponibilité alimentaire (Kcal/personne/jour), Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité de matière grasse en quantité (g/personne/jour), Disponibilité de protéines en quantité (g/personne/jour), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Nourriture, Pertes, Production, Semences, Traitement, Variation de stock]
Index: []

In [27]:
DispoAlimentaire.describe()

#Détection des outliers

Aliments pour animaux  Autres Utilisations  \
count            2720.000000          5496.000000   
mean              479.501838           157.391376   
std              4240.119637          5076.785816   
min                 0.000000             0.000000   
25%                 0.000000             0.000000   
50%                 4.000000             0.000000   
75%                74.000000             4.000000   
max            150000.000000        347309.000000   

       Disponibilité alimentaire (Kcal/personne/jour)  \
count                                    14241.000000   
mean                                        34.789832   
std                                        107.287655   
min                                        -21.000000   
25%                                          0.000000   
50%                                          4.000000   
75%                                         21.000000   
max                                       1711.000000   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                       14015.000000        
mean                                            8.719368        
std                                            24.618223        
min                                            -1.930000        
25%                                             0.060000        
50%                                             0.830000        
75%                                             5.190000        
max                                           430.760000        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                       11794.000000               
mean                                            1.283111               
std                                             3.680399               
min                                            -0.030000               
25%                                             0.010000               
50%                                             0.080000               
75%                                             0.630000               
max                                            60.760000               

       Disponibilité de protéines en quantité (g/personne/jour)  \
count                                       11561.000000          
mean                                            1.223608          
std                                             3.598686          
min                                            -0.370000          
25%                                             0.010000          
50%                                             0.100000          
75%                                             0.660000          
max                                            54.970000          

       Disponibilité intérieure  Exportations - Quantité  \
count              15382.000000             12226.000000   
mean                 640.293460               110.596925   
std                 9067.267153              1053.318990   
min                -3430.000000               -41.000000   
25%                    0.000000                 0.000000   
50%                    7.000000                 0.000000   
75%                   76.750000                 9.000000   
max               739267.000000             42797.000000   

       Importations - Quantité     Nourriture        Pertes     Production  \
count             14852.000000   14015.000000   4278.000000    9180.000000   
mean                 87.264543     347.931359    106.053763    1090.379085   
std                 717.372714    4475.704458   1113.100416   12067.344094   
min                -201.000000    -246.000000      0.000000       0.000000   
25%                   0.000000       0.000000      0.000000       2.000000   
50%                   2.000000       5.000000      4.000000      22.000000   
75%                  18.000000      52.000000     26.000000     191.250000   
max        

In [28]:
DispoAlimentaire.isnull().sum()

#Vérification des valeurs manquantes

Zone                                                                 0
Produit                                                              0
Origine                                                              0
Aliments pour animaux                                            12885
Autres Utilisations                                              10109
Disponibilité alimentaire (Kcal/personne/jour)                    1364
Disponibilité alimentaire en quantité (kg/personne/an)            1590
Disponibilité de matière grasse en quantité (g/personne/jour)     3811
Disponibilité de protéines en quantité (g/personne/jour)          4044
Disponibilité intérieure                                           223
Exportations - Quantité                                           3379
Importations - Quantité                                            753
Nourriture                                                        1590
Pertes                                                           11327
Produc

**Conclusion:**

- 15605 lignes, 18 colonnes
- Les types de données sont corrects
- Pas de doublons
- Beaucoup de valeurs manquantes dans les différentes colonnes

**A corriger:**

- Renommer la colonne "Zone"
- Remplacer les valeurs manquantes par 0

## Préparation des données

### Population

In [29]:
population['Valeur'] = population['Valeur'] * 1000

#Conversion des unités de population en millions d'habitants

In [30]:
population = population.rename(columns={'Zone': 'Pays', 'Valeur': 'Population'})
population.head()

#Modification des noms des colonnes

Pays  Année  Population
0  Afghanistan   2013  32269589.0
1  Afghanistan   2014  33370794.0
2  Afghanistan   2015  34413603.0
3  Afghanistan   2016  35383032.0
4  Afghanistan   2017  36296113.0

### Aide Alimentaire

In [31]:
AideAlimentaire = AideAlimentaire.rename(columns={'Valeur': 'Quantité', 'Pays bénéficiaire':'Pays'})
AideAlimentaire.head()

#Modification des noms des colonnes

Pays  Année              Produit  Quantité
0  Afghanistan   2013  Autres non-céréales       682
1  Afghanistan   2014  Autres non-céréales       335
2  Afghanistan   2013         Blé et Farin     39224
3  Afghanistan   2014         Blé et Farin     15160
4  Afghanistan   2013             Céréales     40504

### Sous Nutrition

In [32]:
SousNutrition['Valeur'] = SousNutrition['Valeur'].fillna(0)

#Remplacement des 'NaN' par '0' --> 594 lignes

In [33]:
SousNutrition.loc[SousNutrition['Valeur'] == '<0.1', 'Valeur'] = 0.01

#Remplacement des données = '<0.1' par '0.01' --> 120 lignes

In [34]:
SousNutrition['Valeur'] = SousNutrition['Valeur'].astype(float)

#Modification du type de la colonne valeur de object à float

In [35]:
SousNutrition['Valeur'] = SousNutrition['Valeur'] * 1000000

#Conversion des unités de population en millions d'habitants

In [36]:
SousNutrition = SousNutrition.rename(columns={'Zone': 'Pays', 'Valeur': 'Nombre de personnes en sous-alimentation (en millions habitants)'})
SousNutrition.head()

#Modification des noms des colonnes

Pays      Année  \
0  Afghanistan  2012-2014   
1  Afghanistan  2013-2015   
2  Afghanistan  2014-2016   
3  Afghanistan  2015-2017   
4  Afghanistan  2016-2018   

   Nombre de personnes en sous-alimentation (en millions habitants)  
0                                          8600000.0                 
1                                          8800000.0                 
2                                          8900000.0                 
3                                          9700000.0                 
4                                         10500000.0

### Dispo Alimentaire

In [37]:
DispoAlimentaire = DispoAlimentaire.fillna(0)

#Remplacement des 'NaN' par '0'

In [38]:
DispoAlimentaire = DispoAlimentaire.rename(columns={'Zone': 'Pays'})
DispoAlimentaire.head()

#Modification du nom de la colonne "Zone" par "Pays"

Pays                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                               0.00               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                               0.00          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      0.0                      0.0   
1                      41.0                      2.0                     40.0   
2                       2.0                      0.0                      2.0   
3                       0.0                      0.0                      0.0   
4                      82.0                      0.0                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     0.0        53.0       0.0         0.0                 0.0  
1        39.0     2.0         3.0       0.0         0.0                 0.0  
2         2.0     0.0         0.0       0.0         0.0                 0.0  
3         0.0     0.0         0.0       0.0         0.0                 0.0  
4        82.0     0.0         0.0       0.0         0.0                 0.0

In [39]:
DispoAlimentaire['Pays']= DispoAlimentaire['Pays'].replace(['Royaume-Uni','Tchéquie (la)'],["Royaume-Uni de Grande-Bretagne et d'Irlande du Nord",'Tchéquie'])
#Modification des noms pour Royaume-Uni et Tchéquie pour matcher avec les df "Population" et "SousNutrition"

## Affichage des 4 Data Frames après modification

In [40]:
population.head()

Pays  Année  Population
0  Afghanistan   2013  32269589.0
1  Afghanistan   2014  33370794.0
2  Afghanistan   2015  34413603.0
3  Afghanistan   2016  35383032.0
4  Afghanistan   2017  36296113.0

In [41]:
AideAlimentaire.head()

Pays  Année              Produit  Quantité
0  Afghanistan   2013  Autres non-céréales       682
1  Afghanistan   2014  Autres non-céréales       335
2  Afghanistan   2013         Blé et Farin     39224
3  Afghanistan   2014         Blé et Farin     15160
4  Afghanistan   2013             Céréales     40504

In [42]:
SousNutrition.head()

Pays      Année  \
0  Afghanistan  2012-2014   
1  Afghanistan  2013-2015   
2  Afghanistan  2014-2016   
3  Afghanistan  2015-2017   
4  Afghanistan  2016-2018   

   Nombre de personnes en sous-alimentation (en millions habitants)  
0                                          8600000.0                 
1                                          8800000.0                 
2                                          8900000.0                 
3                                          9700000.0                 
4                                         10500000.0

In [43]:
DispoAlimentaire.head()

Pays                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                               0.00               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                               0.00          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      0.0                      0.0   
1                      41.0                      2.0                     40.0   
2                       2.0                      0.0                      2.0   
3                       0.0                      0.0                      0.0   
4                      82.0                      0.0                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     0.0        53.0       0.0         0.0                 0.0  
1        39.0     2.0         3.0       0.0         0.0                 0.0  
2         2.0     0.0         0.0       0.0         0.0                 0.0  
3         0.0     0.0         0.0       0.0         0.0                 0.0  
4        82.0     0.0         0.0       0.0         0.0                 0.0